### DISCUSSION TITLE
Common Avalonia Assets folder shared by projects?

### DISCUSSION THREAD

**Discussion by TangDave** [AvaloniaUI/Avalonia#15760] [Q&A] [answered]
In my C# project file I have:
```
  <ItemGroup>
        <AvaloniaResource Include="Assets\**"/>
  </ItemGroup>
```

I can then successfully access image files at runtime as follows:
`... AssetLoader.Open(new Uri("avares://" + assemblyName + "/Assets/Icons/" + imageFileName)) ...`

However, I'd like to share an Assets folder among many applications - this folder would be independent of the project folder. How would I do this?

**Comment by stevemonaco** [comment, marked as answer]
`AvaloniaResource` is baked into the assembly. You're going to have to load from disk (not through the asset system) instead unless you write your own approach to load a DLL at runtime (eg. dynamic plugin) which contains the resources.

**Reply by TangDave** [reply]
Even if it's baked into the assembly, could it not reference an Assets folder not within the project folder?
For instance, I'm using a linked App.xamal file that is outside of the project folder and shared by multiple projects - isn't this a similar idea?

**Reply by stevemonaco** [reply]
If your other projects have a project/NuGet ref so the assembly with the `AvaloniaResource`s gets built into the project, sure. But that's duplicated per application, not shared.

**Reply by TangDave** [reply]
Thanks for the advice.
